audb/core/load.py

```diff
@@ -746,6 +656,58 @@ def job(table):
     )
 
 
+def filtered_dependencies(
+        name: str,
+        version: str,
+        media: typing.Union[str, typing.Sequence[str]],
+        tables: typing.Union[str, typing.Sequence[str]],
+        cache_root: str = None,
+) -> pd.DataFrame:
+    r"""Filter media by tables.
+
+    Return all media files from ``media``
+    that are mentioned in at least one table
+    from ``tables``.
+    This has to download every table
+    for doing the check.
+
+    Args:
+        name: name of database
+        version: version of database
+        media: media files
+        tables: table IDs
+        cache_root: cache folder where databases are stored.
+            If not set :meth:`audb.default_cache_root` is used
+
+    Returns:
+        filtered dependencies
+
+    """
+    deps = dependencies(name, version=version, cache_root=cache_root)
+    if tables is None and media is None:
+        df = deps._df
+    else:
+        available_media = []
+        tables = filter_tables(tables, deps.table_ids)
+        for table in tables:
+            df = load_table(
+                name,
+                table,
+                version=version,
+                cache_root=cache_root,
+                verbose=False,
+            )
+            available_media += list(
+                df.index.get_level_values('file').unique()
+            )
+
+        media = filter_media(media, deps.media, name, version)
+        available_media = [m for m in media if m in list(set(available_media))]
+        df = deps._df.loc[available_media]
```

Comment:
```suggestion
        df = deps().loc[available_media]
```

Reply:
There is one more usage of `deps._df` within `_files_duration()` in `load.py`. Maybe change that one, too.

Reply:
Here, I'm not sure. If we change back storing dependencies as dictionary `.loc` will anyway not work. So maybe better to stay with `_df` here?

Reply:
No, you are right. I changed it in both places now.